ios/FluentUI/Bottom Sheet/BottomCommandingController.swift

```diff
@@ -0,0 +1,596 @@
+        let iconView = UIImageView(image: item.image)
+        iconView.tintColor = Constants.tableViewIconTintColor
+        cell.setup(title: item.title, subtitle: "", footer: "", customView: iconView, customAccessoryView: nil, accessoryType: .none)
+        cell.bottomSeparatorType = .none
+        cell.isEnabled = item.isEnabled
+        cell.backgroundColor = Constants.tableViewBackgroundColor
+    }
+
+    // Reloads view in place from the given item object
+    private func reloadView(from item: CommandingItem) {
+        guard let binding = itemToBindingMap[item] else {
+            return
+        }
+        let staleView = binding.view
+
+        switch binding.location {
+        case .heroSet:
+            if let stackIndex = heroCommandStack.arrangedSubviews.firstIndex(of: staleView) {
+                removeBinding(binding)
+                let newView = createAndBindHeroCommandView(with: item)
+                staleView.removeFromSuperview()
+                heroCommandStack.insertArrangedSubview(newView, at: stackIndex)
+            }
+        case .list:
+            if let cell = binding.view as? TableViewCell {
+                setupTableViewCell(cell, with: item)
+            }
+        }
+    }
+
+    private var itemToBindingMap: [CommandingItem: ItemBinding] = [:]
+
+    private var viewToBindingMap: [UIView: ItemBinding] = [:]
+
+    private var bottomBarView: UIView?
+
+    private var bottomSheetController: BottomSheetController?
+
+    private var isHeroCommandStackLoaded: Bool = false
+
+    private var isTableViewLoaded: Bool = false
+
+    private var isInSheetMode: Bool { bottomSheetController != nil }
+
+    private var isExpandable: Bool { expandedListSections.count > 0 }
+
+    private var bottomSheetHeroStackTopConstraint: NSLayoutConstraint?
+
+    private var bottomSheetHeroStackTopMargin: CGFloat {
+        isExpandable ? Constants.BottomSheet.heroStackExpandableTopMargin : Constants.BottomSheet.heroStackNonExpandableTopMargin
+    }
+
+    private var bottomSheetHeroStackHeight: CGFloat { Constants.heroButtonHeight + Constants.BottomSheet.heroStackBottomMargin + bottomSheetHeroStackTopMargin }
+
+    private var heroCommandWidthConstraints: [NSLayoutConstraint] {
+        heroItems.compactMap { (itemToBindingMap[$0] as? HeroItemBinding)?.widthConstraint }
+    }
+
+    private enum ItemLocation {
+        case heroSet
+        case list
+    }
+
+    private class ItemBinding {
+        let item: CommandingItem
+        let view: UIView
+        let location: ItemLocation
+
+        init(item: CommandingItem, view: UIView, location: ItemLocation) {
+            self.item = item
+            self.view = view
+            self.location = location
+        }
+    }
+
+    private class HeroItemBinding: ItemBinding {
+        let widthConstraint: NSLayoutConstraint
+
+        init(item: CommandingItem, view: UIView, location: ItemLocation, widthConstraint: NSLayoutConstraint) {
```

Comment:
Not sure how I feel about location and widthConstraint being part of the binding. In my mind "binding" is a simple map and that sort of info lives on either the view or item itself.

Maybe it's just the name "binding" though? Because that also implies some level of functionality tying things together but that functionality seems to live in the usage rather than the implementation?

Reply:
It's more of BindingMetadata. Constraints are external to the item views, and items shouldn't really know how they're represented so that also feels wrong. So they should be somewhere in this controller IMO and this was a nice way to keep them tied to a binding.

Reply:
Maybe BindingMetadata is better? Binding makes me think of what that word means in SwiftUI and the scope of this is beyond that